resources/session04/mashup.py

```diff
@@ -0,0 +1,213 @@
+        new_label = clean_data(key_cell)
+        current_label = new_label if new_label else current_label
+        rdata.setdefault(current_label, []).append(clean_data(val_cell))
+    return rdata
+
+
+def is_inspection_data_row(elem):
+    is_tr = elem.name == 'tr'
+    if not is_tr:
+        return False
+    td_children = elem.find_all('td', recursive=False)
+    has_four = len(td_children) == 4
+    this_text = clean_data(td_children[0]).lower()
+    contains_word = 'inspection' in this_text
+    does_not_start = not this_text.startswith('inspection')
+    return is_tr and has_four and contains_word and does_not_start
+
+
+def get_score_data(elem):
+    inspection_rows = elem.find_all(is_inspection_data_row)
+    samples = len(inspection_rows)
+    total = 0
+    high_score = 0
+    average = 0
+    for row in inspection_rows:
+        strval = clean_data(row.find_all('td')[2])
+        try:
+            intval = int(strval)
+        except (ValueError, TypeError):
+            samples -= 1
+        else:
+            total += intval
+            high_score = intval if intval > high_score else high_score
+
+    if samples:
+        average = total/float(samples)
+    data = {
+        u'Average Score': average,
+        u'High Score': high_score,
+        u'Total Inspections': samples
+    }
+    return data
+
+def add_color(inspection_data):
+    '''
+    Add color and symbols for high scores
+    '''
+    if inspection_data[u'Average Score'] > 50:
+        color = 'ff0000'
+    elif inspection_data[u'Average Score'] > 30:
+        color = 'ffff00'
+    else:
+        color = '33cc33'
+    if inspection_data[u'High Score'] > 80:
+        size = 'large'
+    else:
+        size = 'medium'
+    return({u'marker-color': color,
+            u'marker-size': size})
+
+    
+def sorted_results(sort_type, count, reverse_flag):
+    '''
+    Return list of dictionaries sorted in order of average score.
+    sort_type - field to sort on
+    count - number of entries returned
+    reverse - boolean to indicate results are returned in reverse sorted order.
+    '''
+    result=[]
+    use_params = {
+        'Inspection_Start': '2/1/2013',
+        'Inspection_End': '2/1/2015',
+        'Zip_Code': '98101'
+    }
+    # html = get_inspection_page(**use_params)
+    html = load_inspection_page('inspection_page.html')
+    parsed = parse_source(html)
+    content_col = parsed.find("td", id="contentcol")
+    data_list = restaurant_data_generator(content_col)
+    for data_div in data_list:
+        metadata = extract_restaurant_metadata(data_div)
+        inspection_data = get_score_data(data_div)
+        metadata.update(inspection_data)
+        color_data = add_color(inspection_data)
+        metadata.update(color_data)
+        result.append(metadata)
+    result.sort(key=operator.itemgetter(SORTING_TYPES[sort_type]),
+            reverse=not reverse_flag)
+    return(result[:count])
```

Comment:
the parens here are non-stylistic.  `return` is a statement and does not use parens.